tests/templates/test_subroutines.py

```diff
@@ -1207,3 +1208,263 @@ def circuit():
             return [qml.expval(qml.PauliZ(wires=i)) for i in range(n_wires)]
 
         assert np.allclose(circuit(), expectation)
+
+
+class TestPermute:
+    """Tests for the Permute template from the pennylane.templates.subroutine module."""
+
+    @pytest.mark.parametrize(
+        "permutation_order,expected_error_message",
+        [
+            ([0], "Permutations must involve at least 2 qubits."),
+            ([0, 1, 2], "Permutation must specify outcome of all wires."),
+            ([0, 1, 1, 3], "Values in a permutation must all be unique"),
+            ([4, 3, 2, 1], r"not present in wire set"),
+        ],
+    )
+    def test_invalid_inputs_qnodes(self, permutation_order, expected_error_message):
+        """Tests if errors are thrown for invalid permutations with QNodes."""
+
+        dev = qml.device("default.qubit", wires=4)
+
+        @qml.qnode(dev)
+        def permute_qubits():
+            Permute(permutation_order, wires=dev.wires)
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match=expected_error_message):
+            permute_qubits()
+
+    @pytest.mark.parametrize(
+        "permutation_order,expected_error_message",
+        [
+            ([0], "Permutations must involve at least 2 qubits."),
+            ([2, "c", "a", 0], "Permutation must specify outcome of all wires."),
+            ([2, "a", "c", "c", 1], "Values in a permutation must all be unique"),
+            ([2, "a", "d", "c", 1], r"not present in wire set"),
+        ],
+    )
+    def test_invalid_inputs_tape(self, permutation_order, expected_error_message):
```

Comment:
What is the motivation for testing on both the QNode and tape level?

Reply:
Just to cover both cases in the interim, and for the cases where wires are labelled with strings (I didn't think that was possible with QNodes?). Plus once things transition over to tape mode, we already have the tests written and can remove the QNode tests.

Reply:
> I didn't think that was possible with QNodes?

It should be possible 🤔

Reply:
Oh I didn't know that! Awesome :)